pkg/http/handler/timeout.go

```diff
@@ -43,40 +52,33 @@ import (
 // https://golang.org/pkg/net/http/#Handler.
 //
 // The implementation is largely inspired by http.TimeoutHandler.
-func TimeToFirstByteTimeoutHandler(h http.Handler, dt time.Duration, msg string) http.Handler {
-	return &timeoutHandler{
-		handler: h,
-		body:    msg,
-		timeout: dt,
+func TimeToFirstByteTimeoutHandler(h http.Handler, msg string, timeoutFunc timeoutFunc) http.Handler {
+	return &timeToFirstByteTimeoutHandler{
+		handler:     h,
+		body:        msg,
+		timeoutFunc: timeoutFunc,
 	}
 }
 
-type timeoutHandler struct {
-	handler http.Handler
-	body    string
-	timeout time.Duration
-}
-
-func (h *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (h *timeToFirstByteTimeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, cancel := context.WithCancel(r.Context())
 	defer cancel()
 
 	// done is closed when h.handler.ServeHTTP completes and contains
 	// the panic from h.handler.ServeHTTP if h.handler.ServeHTTP panics.
 	done := make(chan interface{})
-
-	tw := &timeoutWriter{w: w}
+	twriter := &timeoutWriter{w: w}
```

Comment:
don't know if it matters very much, but now `tw` is back to `h` in the receiver name, we don't need this rename any more (and it's a lightly used local variable so conciseness is ok..). If you feel strongly that it's clearer the longer way though I don't think it's very important.